src/main/javascript/severities.js

```diff
@@ -307,6 +309,35 @@ class SeveritiesPage extends React.Component {
         this.displayPlot(airframe);
     }
 
+    getDescription(eventName) {
+        var text = "";
+
+        $.ajax({
+                type: 'GET',
+                url: '/protected/get_event_description',
+                data : {eventName : eventName},
+                dataType : 'json',
+                success : function(response) {
+                    console.log("received response: " + response);
+
+                    $('#loading').hide();
+
+                    if (response.err_msg) {
+                        errorModal.show(response.err_title, response.err_msg);
+                    }
+
+                    text = response + "";
+                },
+                error : function(jqXHR, textStatus, errorThrown) {
+                    errorModal.show("Error Getting Event Description", errorThrown);
+                },
+                async: false
+            });
+
+        console.log("returning text: " + text);
+        return text;
```

Comment:
I would add a set/map in the `state` that contains the description. You can create a set that has keys as the event name that map to their respective event description. Then you can just check to see if it exists - if not we would query the server and populate that key. This would make the class cache the descriptions it already has asked the server for, cutting down on network resources.

Reply:
Yup this would be great, i don't think we want to be pinging the server everytime we mouseover a tooltip (especailly if we've already grabbed it).